src/diffusers/models/unet_2d_condition.py

```diff
@@ -571,8 +572,17 @@ def __init__(
                 positive_len = cross_attention_dim
             elif isinstance(cross_attention_dim, tuple) or isinstance(cross_attention_dim, list):
                 positive_len = cross_attention_dim[0]
+
             self.position_net = PositionNet(positive_len=positive_len, out_dim=cross_attention_dim)
```

Comment:
Above replace:
```py
if attention_type == "gated":
```
with
```py
if attention_type in ["gated", "gated-text-image"]:
```
```suggestion
            self.position_net = PositionNet(positive_len=positive_len, out_dim=cross_attention_dim, feature_type="text" if attention_type == "gated" else "text-image")
```